tests/Oryx.Tests.Common/EndToEndTestHelper.cs

```diff
@@ -187,63 +188,28 @@ public static class EndToEndTestHelper
                     runCmd,
                     runArgs);
 
-                await RunAssertsAsync(
-                    () =>
-                    {
-                        // An exception could have occurred when a Docker process failed to start.
-                        Assert.Null(runResult.Exception);
-                        Assert.False(runResult.Process.HasExited);
-                        return Task.CompletedTask;
-                    },
-                    runResult,
-                    output);
+                // An exception could have occurred when a Docker process failed to start.
+                Assert.Null(runResult.Exception);
+                Assert.False(runResult.Process.HasExited);
 
                 for (var i = 0; i < MaxRetryCount; i++)
                 {
                     await Task.Delay(TimeSpan.FromSeconds(DelayBetweenRetriesInSeconds));
 
                     try
                     {
-                        // Make sure the process is still alive and fail fast if not alive.
-                        await RunAssertsAsync(
```

Comment:
do we still need this method? If yes, doesn't it have the same problem?

Reply:
We use this method for the scenario when doing a 'build' where we kick off a process and wait for it to exit and write out the output when asserts fail. Since the process already exits, we do not have an issue here.